service/matching/pri_matcher.go

```diff
@@ -324,7 +331,7 @@ func (tm *priTaskMatcher) forwardPolls() {
 func (tm *priTaskMatcher) Offer(ctx context.Context, task *internalTask) (bool, error) {
 	finish := func() (bool, error) {
 		res, ok := task.getResponse()
-		bugIf(!ok, "Offer must be given a sync match task")
+		softassert.That(tm.logger, ok, "expected a sync match task")
```

Comment:
needs to return error?

Reply:
Seems like `return false, nil` has the same effect as an error.